activerecord/lib/active_record/database_configurations/database_config.rb

```diff
@@ -28,6 +22,12 @@ def new_connection
         adapter_class.new(configuration_hash)
       end
 
+      def validate
```

Comment:
Naming is hard. This could be `validate!` since it raises on invalid, rather than returning false.